python/lsst/alert/stream/alertConsumer.py

```diff
@@ -63,17 +72,18 @@ def poll(self, decode=False, verbose=True):
         verbose : `boolean`
             If True, returns every message. If False, only raises EopError.
         """
-        msg = self.consumer.poll()
-
-        if msg.error():
-            raise EopError(msg)
-        else:
-            if verbose is True:
-                if decode is True:
-                    return self.decodeMessage(msg)
-                else:
-                    ast_msg = literal_eval(str(msg.value(), encoding='utf-8'))
-                    return ast_msg
+        msg = self.consumer.poll(timeout=1)
```

Comment:
If the timeout is 1 second, will it close before the next visit if I send alert bursts that are 39 seconds apart?  Can I change this?  Or no, is that not how it will work?  I read the docs, but I am not sure.

Reply:
You should test this to make sure I'm right, but...

The point is that the `consumer.poll()` method is blocking. When you call it, everything else stops and waits for it to return. That means that if something else happens (like a user hitting `ctrl+c`) it won't get noticed until after `poll()` has finished executing.

If you call `poll()` and no data is received, it will block forever.

If you call `poll(timeout=X)`, then after `X` seconds of receiving no data, it will return `None`. Your `AlertConsumer.poll()` function then immediately returns nothing at line 87 (the `if` at line 77 returns `False`).

This nothing gets sent back to line 104 of `filterStream.py`, and that code continues executing. In particular, if a `KeyboardInterrupt` has been received, it can now be caught by the exception handler at line 118. If no such interrupt has happened, we just continue with the `while True` loop (line 102), which immediately calls `poll()` again.

Sorry; that's a lot of words. Here's an attempt to be shorter:

Previously, the code was blocking until data was received, which is forever if there's no data.

Now, the code will block for 1 second, wake up and check for `KeyboardInterrupt`, then go back and block for another second.

Does that make sense?

Reply:
Yes, that makes sense, thanks.  I could never get Keyboard Interrupt to work, I always just killed the container...

Reply:
Thanks, yes that works fine.